File: intro/spring/bean/configuration/src/main/java/org/tsdes/intro/spring/bean/configuration/Application.java
package org.tsdes.intro.spring.bean.configuration;

import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

/**
 * Created by arcuri82 on 26-Jan-18.
 */
@SpringBootApplication
public class Application {

    /*
        Note: @SpringBootApplication brings in @Configuration as well.
        Annotations can have annotations as well
     */

    @Bean
    public NotAService notAServiceFoo(){
        return new NotAService("foo");
    }

    @Bean
    public NotAService notAServiceBar(){
        return new NotAService("bar");
    }
}

File: spring/security/database/src/main/kotlin/org/tsdes/spring/security/database/Application.kt
package org.tsdes.spring.security.database

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.ComponentScan

/**
 * Created by arcuri82 on 08-Nov-17.
 */
@SpringBootApplication
@ComponentScan
class Application


fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}

